app.js

```diff
@@ -74,30 +84,28 @@ app.post('/api/request', function(req, res) {
     var decision = req.body ? req.body.decision : null;
     var isManifest = req.body ? req.body.isManifest === 'true' : false;
     var token = req.body ? req.body.token : null;
+    var id = uuid.v4();
 
     if (!url || !decision || !token) {
-        res.status(500).send({error: 'Missing parameters {url, decision, token}.'});
+        res.status(500).send('Missing required parameters “url”, “decision” and/or “token”.');
     }
     else {
-        if (requests[url]) {
-            res.send('Spec at ' + url + ' is yet pending validation OR has been already submitted. Check “/api/status”.');
-        }
-        else {
-            requests[url] = {
-                'url': url,
-                'decision': decision,
-                'isManifest': isManifest,
-                'token': token,
-                'jobs': {},
-                'history': new History()
-            };
-            orchestrate(requests[url], isManifest, token).then(function () {
-                console.log('Spec at ' + url + ' (decision: ' + decision + ') has FINISHED.');
-            }, function (err) {
-                console.log('Spec at ' + url + ' (decision: ' + decision + ') has FAILED.');
-            });
-            res.send('Spec at ' + url + ' (decision: ' + decision + ') added to the queue.');
-        }
+        requests[id] = {
+            'id': id,
+            'url': url,
+            'decision': decision,
+            'isManifest': isManifest,
+            'jobs': {},
+            'history': new History(),
+            'status': STATUS_STARTED
+        };
+
+        orchestrate(requests[id], isManifest, token).then(function () {
+            console.log('Spec at ' + url + ' (decision: ' + decision + ') has FINISHED.');
+        }, function (err) {
+            console.log('Spec at ' + url + ' (decision: ' + decision + ') has FAILED.');
+        });
+        res.status(202).send(id);
```

Comment:
202 mean that you accepted the job, but don't give information about the status.
303 redirect to the status, in that case you will see the status (starting with the successful creation, of course).